app/controllers/forest/companies_controller.rb

```diff
@@ -43,4 +43,32 @@ def upload_legal_docs
     # Once the upload is finished, send a success message to the admin user in the UI.
     render json: { success: 'Legal documents are successfully uploaded.' }
   end
+
+  def new_emitted_transaction
+    
+    # here is the logic behind the smart action form
+    attrs = params.dig('data','attributes', 'values')
+    beneficiary_company_id = attrs['Beneficiary company']
+    emitter_company_id = params.dig('data','attributes')["ids"][0]
+    amount = attrs['Amount']
+    Transaction.create!(
+      emitter_company_id: emitter_company_id,
+      beneficiary_company_id: beneficiary_company_id,
+      beneficiary_iban: Faker::Code.imei,
+      emitter_iban: Faker::Code.imei,
+      vat_amount: Faker::Number.number(4),
+      fee_amount: Faker::Number.number(4),
+      status: ['to_validate', 'validated', 'rejected'].sample,
+      note: Faker::Lorem.paragraph,
+      amount: amount,
+      emitter_bic: Faker::Code.nric,
+      beneficiary_bic: Faker::Code.nric
+    )
+
+    #success message and refresh cache
```

Comment:
The comment is not explicit and this is the main purpose of this example. You should explain in details what you're doing here in the `refresh` option

Reply:
And "refresh cache" is not the right wording, as there is no cache (same thing for the PR title).